registry/tags.go

```diff
@@ -8,10 +8,18 @@ func (registry *Registry) Tags(repository string) ([]string, error) {
 	url := registry.url("/v2/%s/tags/list", repository)
 	registry.Logf("registry.tags url=%s repository=%s", url, repository)
 
+	tags := []string{}
+
 	var response tagsResponse
-	if err := registry.getJson(url, &response); err != nil {
-		return nil, err
+	var err error
+
+	for url != "" {
```

Comment:
I'm not a big fan of relying on an empty string to know when we need to stop the loop.
How about doing it in a similar way as [Reader#Read](https://golang.org/pkg/bufio/#Reader.Read), with an `EOF` error that we consider differently from other errors?